File: 2019/day_2/part_1.py
with open('input.txt') as f:
    file_contents = f.read()

instructions = file_contents.split(',')
instructions = [int(i) for i in instructions]

instructions[1] = 12
instructions[2] = 2

position = 0
while 1:
    opcode = instructions[position]
    if opcode == 99:
        break

    a = instructions[position + 1]
    b = instructions[position + 2]
    dest = instructions[position + 3]

    result = None
    if opcode == 1:
        result = instructions[a] + instructions[b]
    elif opcode == 2:
        result = instructions[a] * instructions[b]

    instructions[dest] = result
    position += 4

print(instructions)

File: 2022/day_02/part_1.rs
use std::fs::File;
use std::io::{self, BufRead};

#[derive(Debug, Eq, PartialEq)]
enum Throw {
    Rock,
    Paper,
    Scissors,
}

#[derive(Debug)]
struct Round {
    opponent: Throw,
    response: Throw,
}

fn string_to_throw(s: &str) -> Throw {
    match s {
        "A" => Throw::Rock,
        "B" => Throw::Paper,
        "C" => Throw::Scissors,
        "X" => Throw::Rock,
        "Y" => Throw::Paper,
        "Z" => Throw::Scissors,
        _ => panic!("Not a valid Throw code!"),
    }
}

fn calc_round_score(round: &Round) -> i32 {
    let mut result: i32 = match round.response {
        Throw::Rock => 1,
        Throw::Paper => 2,
        Throw::Scissors => 3,
    };
    
    if round.opponent == round.response {
        result += 3;
    } else if 
        (round.opponent == Throw::Rock && round.response == Throw::Paper)
        || (round.opponent == Throw::Paper && round.response == Throw::Scissors)
        || (round.opponent == Throw::Scissors && round.response == Throw::Rock)
    {
        result += 6
    }
    return result;
}

fn main() {
    let file = File::open("input.txt").unwrap();
    let lines = io::BufReader::new(file).lines();

    let mut total_score = 0;

    for line_result in lines {
        let line = line_result.unwrap();
        let mut parts = line.split(" ");
        let opp = string_to_throw(parts.next().unwrap());
        let res = string_to_throw(parts.next().unwrap());

        let round = Round { opponent: opp, response: res };
        let round_score = calc_round_score(&round);

        total_score += round_score;
        // println!("{:?} -----> {:?}", round, round_score);
    }

    println!("{total_score}");
}
